janusgraph-test/src/test/java/org/janusgraph/graphdb/serializer/SerializerGraphConfiguration.java

```diff
@@ -89,5 +93,34 @@ public void testOnlyRegisteredSerialization() {
         }
     }
 
+    @Test
+    public void testCustomHashMapSerialization() {
+        JanusGraphManagement mgmt = graph.openManagement();
+        PropertyKey time = mgmt.makePropertyKey("time").dataType(Integer.class).make();
```

Comment:
If you just want to test your custom attribute registration I think you could add a test to SerializerTest. In this case I agree your test wouldn't include a graph at all and would be more like [SerializerTest#enumSerializeTest](https://github.com/JanusGraph/janusgraph/blob/cb9c06aeb8ab95c3a4c4cc7575b7ad97ff79ddca/janusgraph-test/src/test/java/org/janusgraph/graphdb/serializer/SerializerTest.java#L225-L236) (replacing enum test data/asserts with a HashMap). For the test as you have it written this looks like a good place for it but I'm not sure it runs by default under surefire (the class name doesn't end in "Test" and I don't see it in any test includes).